File: pc/app/java/net/qaul/libqaul/LibQaul.java
package net.qaul.libqaul;

public class LibQaul {
    
    static {
        System.loadLibrary("libqaul");

        System.out.println("Loaded libqaul library.");
    }

    public static native void hello();

    public static void receiveRequest() {
        System.out.println("Called From FFI");
    }

}

File: kotlin/app/src/main/kotlin/net/qaul/libqaul/libqaul.kt
package net.qaul.libqaul

/// hello message from lib qaul
// dummy function for testing
external fun hello(): String

/// send an SYS message from BLE library to libqaul
external fun syssend(message: ByteArray)

/// receive an SYS message from libqaul to BLE library
external fun sysreceive(): ByteArray

/// load rust libqaul shared library
fun loadLibqaul() {
    System.loadLibrary("libqaul")
}